src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+                    {
+                        string referencingColumn = fkDefinition.ReferencingColumns[idx];
+                        string referencedColumn = fkDefinition.ReferencedColumns[idx];
+
+                        // The input data for current entity should not specify a value for a referencing column -
+                        // as it's value will be derived from the insertion in the referenced (target) entity.
+                        if (derivableColumnsFromRequestBody.TryGetValue(referencingColumn, out string? referencingColumnSource))
+                        {
+                            string conflictingSource;
+                            if (referencingColumnSource.StartsWith(relationshipSourceIdentifier))
+                            {
+                                // If the source name starts with "$", this indicates the source for the referencing column
+                                // was another relationship.
+                                conflictingSource = "Relationship: " + referencingColumnSource.Substring(relationshipSourceIdentifier.Length);
+                            }
+                            else
+                            {
+                                conflictingSource = referencingColumnSource.Equals(parentEntityName) ? $"Parent entity: {referencingColumnSource}" : $"entity: {entityName}";
+                            }
+
+                            metadataProvider.TryGetExposedColumnName(entityName, referencingColumn, out string? exposedColumnName);
+                            throw new DataApiBuilderException(
+                                message: $"Found conflicting sources of values for the field: {exposedColumnName} for entity: {entityName} at level: {nestingLevel}." +
+                                $"Source 1: {conflictingSource}, Source 2: Relationship: {relationshipName}.",
+                                statusCode: HttpStatusCode.BadRequest,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                        }
+
+                        // Case 2: When a column whose value is to be derived from the insertion in current entity
+                        // (happens when the parent entity is a referencing entity in a relationship with current entity),
+                        // is a referencing column in the current relationship, we pass on the responsibility of getting the value
+                        // of such a column to the target entity in the current relationship.
+                        if (columnsToDerive.Contains(referencingColumn))
+                        {
+                            // We optimistically assume that we will get the value of the referencing column
+                            // from the insertion in the target entity.
+                            columnsToDerive.Remove(referencingColumn);
+                        }
```

Comment:
This is when the parent entity turns out to be the referencing entity. But, we already know that the current entity is the referencing entity ---> L322.

So, can we move this out of the `if` block --> block that checks that the current entity is the referencing entity

Reply:
This is the case when parent is referencing child and child is referencing its child.

A->B->C. I have added an example here: https://github.com/Azure/data-api-builder/blob/e0fbd36687017f4a289d3143bbd09364f94d325d/src/Core/Services/MultipleMutationInputValidator.cs#L218. The code block which you have highlighted specifically covers this case.